machine/fdt.c

```diff
@@ -204,6 +221,56 @@ void query_mem(uintptr_t fdt)
   assert (mem_size > 0);
 }
 
+///////////////////////////////////////////// ROOT SCAN /////////////////////////////////////////
+
+struct root_scan {
+  const struct fdt_scan_node *root;
+  const uint32_t *version_value;
+  int version_len;
+};
+
+static void root_open(const struct fdt_scan_node *node, void *extra)
+{
+  struct root_scan *scan = (struct root_scan *)extra;
+  if (!scan->root) {
+    scan->root = node;
+  }
+}
+
+static void root_prop(const struct fdt_scan_prop *prop, void *extra)
+{
+  struct root_scan *scan = (struct root_scan *)extra;
+  if (!strcmp(prop->name, "sri-cambridge,version") && !scan->version_value) {
```

Comment:
I generally find `strcmp(a, b) == 0` to be easier to parse (the ! can be missed quite easily), but don't really mind either way.

Reply:
I definitely agree, but there are 10s of other strcmps in the file done this way, so I figured I'd match the style.